internal/k8s/validation.go

```diff
@@ -223,30 +367,94 @@ func validatePlusOnlyAnnotation(context *annotationValidationContext) field.Erro
+	allErrs := field.ErrorList{}
+	if !context.appProtectEnabled {
+		return append(allErrs, field.Forbidden(context.fieldPath, "annotation requires AppProtect"))
+	}
+	return allErrs
+}
+
+func validateInternalRoutesOnlyAnnotation(context *annotationValidationContext) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if !context.internalRoutesEnabled {
+		return append(allErrs, field.Forbidden(context.fieldPath, "annotation requires Internal Routes enabled"))
+	}
+	return allErrs
+}
+
 func validateBoolAnnotation(context *annotationValidationContext) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if _, err := configs.ParseBool(context.value); err != nil {
-		return append(allErrs, field.Invalid(context.fieldPath, context.value, "must be a valid boolean"))
+		return append(allErrs, field.Invalid(context.fieldPath, context.value, "must be a boolean"))
 	}
 	return allErrs
 }
 
 func validateTimeAnnotation(context *annotationValidationContext) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if _, err := configs.ParseTime(context.value); err != nil {
-		return append(allErrs, field.Invalid(context.fieldPath, context.value, "must be a valid time"))
+		return append(allErrs, field.Invalid(context.fieldPath, context.value, "must be a time"))
 	}
 	return allErrs
 }
 
-func validateNonNegativeIntAnnotation(context *annotationValidationContext) field.ErrorList {
+func validateUint64Annotation(context *annotationValidationContext) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if _, err := configs.ParseUint64(context.value); err != nil {
 		return append(allErrs, field.Invalid(context.fieldPath, context.value, "must be a non-negative integer"))
 	}
 	return allErrs
 }
 
+func validateInt64Annotation(context *annotationValidationContext) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if _, err := configs.ParseInt64(context.value); err != nil {
+		return append(allErrs, field.Invalid(context.fieldPath, context.value, "must be an integer"))
+	}
+	return allErrs
+}
+
+func validateIntAnnotation(context *annotationValidationContext) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if _, err := configs.ParseInt(context.value); err != nil {
+		return append(allErrs, field.Invalid(context.fieldPath, context.value, "must be an integer"))
+	}
+	return allErrs
+}
+
+func validatePortListAnnotation(context *annotationValidationContext) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if _, err := configs.ParsePortList(context.value); err != nil {
+		return append(allErrs, field.Invalid(context.fieldPath, context.value, "must be a comma-separated list of port numbers"))
+	}
+	return allErrs
+}
+
+func validateServiceListAnnotation(context *annotationValidationContext) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if _, err := configs.ParseServiceList(context.value); err != nil {
+		return append(allErrs, field.Invalid(context.fieldPath, context.value, "must be a comma-separated list of services"))
```

Comment:
for this and similar annotations, would it make sense to check that services are also referenced in the spec?

Reply:
Good idea, just added that in now.